inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -267,4 +267,89 @@ namespace inlanecruising_plugin
 
         return trajectory;
     }
+
+    tk::spline InLaneCruisingPlugin::compute_fit(std::vector<lanelet::BasicPoint2d> basic_points){
+        int n = int(basic_points.size())/3;
+
+        tk::spline spl;
+
+        std::vector<double> points_x;
+        points_x.push_back(basic_points[n].x());
+        points_x.push_back(basic_points[2*n].x());
+        points_x.push_back(basic_points[3*n].x());
+
+        std::vector<double> points_y;
+        points_y.push_back(basic_points[n].y());
+        points_y.push_back(basic_points[2*n].y());
+        points_y.push_back(basic_points[3*n].y());
+
+        if (points_x.size()<3 || points_y.size()<3){
+            throw std::invalid_argument("Insufficient Spline Points");
+        }
+
+        spl.set_points(points_x, points_x);
+
+        return spl;
+
+    }
+
+    std::vector<double> InLaneCruisingPlugin::compute_orientation_from_fit(tk::spline curve, std::vector<double> sampling_points){
+        std::vector<double> orientations;
+        std::vector<double> cur_point{0.0, 0.0};
+        std::vector<double> next_point{0.0, 0.0};
+        double lookahead = 0.3;
+        for (size_t i=0; i<sampling_points.size(); i++){
+            cur_point[0] = sampling_points[i];
+            cur_point[1] = curve(cur_point[0]);
+            next_point[0] = cur_point[0] + lookahead;
+            next_point[1] = curve(next_point[0]);
+            double res = calculate_yaw(cur_point, next_point);
+            orientations.push_back(res);
```

Comment:
What's the conclusion on the reference frame of sampling_points here? Is it still X in the map frame? Won't that fail when there is multiple possible x values on the curve like we discussed in slack yesterday?